pallets/liquid-staking-two/src/lib.rs

```diff
@@ -247,45 +249,116 @@ pub mod pallet {
             origin: OriginFor<T>,
             era_index: EraIndex,
         ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?; //TODO(Alan): Who can call this extrinsic?
+            CurrentEra::<T>::put(era_index);
+            // TODO(Alan): Should we deposit event?
             Ok(().into())
         }
 
-        //todo，record reward on each era, invoked by stake-client
+        /// Invoked by offchain stake client. Declare reward received.
+        ///
+        /// `origin` should be approved account.(not implement yet).
+        ///
+        /// Q: What's the agent?
+        /// Q: Should update exchange_rate?
         #[pallet::weight(10_000)]
         #[transactional]
-        pub fn record_reward(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-            Ok(().into())
-        }
-
-        //todo invoked by stake-client, considering insurrance pool
-        #[pallet::weight(10_000)]
-        #[transactional]
-        pub fn record_slash(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-            Ok(().into())
-        }
-
-        // bond/unbond/rebond/bond_extra may be merge into one
-        #[pallet::weight(10_000)]
-        #[transactional]
-        pub fn record_bond_response(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-            Ok(().into())
-        }
-
-        #[pallet::weight(10_000)]
-        #[transactional]
-        pub fn record_bond_extra_response(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+        pub fn record_reward(
+            origin: OriginFor<T>,
+            agent: T::AccountId,
+            #[pallet::compact] amount: Balance,
+            block_number: BlockNumber,
+        ) -> DispatchResultWithPostInfo {
+            let _sender = ensure_signed(origin)?;
+            MatchingPool::<T>::try_mutate(
+                Self::current_era(),
+                |matching_buffer| -> DispatchResult {
+                    let total_stake_amount = matching_buffer
+                        .total_stake_amount
+                        .checked_add(amount)
+                        .ok_or(ArithmeticError::Overflow)?;
+                    *matching_buffer = MatchingBuffer {
+                        total_stake_amount,
+                        ..matching_buffer.clone()
+                    };
+                    Ok(().into())
+                },
+            )?;
+
+            StakingOperationHistory::<T>::insert(
```

Comment:
in every era, we can only have one record_reward operation, so I think it's better to check if it already exists.

and then, we insert it into operation history.